docs/csharp/programming-guide/classes-and-structs/local-functions.md

```diff
@@ -80,22 +80,39 @@ Like with the method iterator, you can refactor the preceding example and put th
+### Typing
 
-Third, the compiler can perform static analysis that enables local functions to definitely assign captured variables in the enclosing scope. Consider this example:
+Lambda expressions rely on the typing of the `Func`/`Action` variable that they're assigned to determine the argument and return types. In local functions, since the syntax is much like writing a normal method, argument types and return type are already part of the function declaration.
+
+### Definite Assignment
+
+Local functions have different rules for definite assignment than lambda expressions. A local function declaration can be referenced from any code location where it is in scope; in our first example `LocalFunctionFactorial`, we could declare our local function either above or below the `return` statement and not trigger any compiler errors.
+
+Lambda expressions follow the rules for definite assignment more closely. In order for a lambda expression to be used, the `Func`/`Action` variable that it will be assigned to must be declared and the lambda expression assigned to it. Notice that `LambdaFactorial` must declare and initialize the lambda expression `nthFactorial` before defining it. Not doing so results in a compile time error for referencing `nthFactorial` before assigning it.
+
+These differences mean that recursive algorithms are easier to create using local functions. You can declare and define a local function that calls itself. Lambda expressions must be declared, and assigned a default value before they can be re-assigned to a body that references the same lambda expression.
+
+### Conversion to a Delegate
+
+Both local functions and lambda expressions require that any captured variables are definitely assigned when either is converted to a delegate, but *when* they're converted differs. Lambda expressions are converted to delegates when they're declared, while local functions are only converted to delegates when ***used*** as a delegate.
+
+If you declare a local function and only reference it by calling it like a method, it will not be converted to a delegate. That rule enables you to declare a local function at any convenient location in its enclosing scope. It's common to declare local functions at the end of the parent method, after any return statements.
```

Comment:
```suggestion
If you declare a local function and only reference it by calling it like a method, it will not be converted to a delegate. That rule enables you to declare a local function at any convenient location in its enclosing scope. It's common to declare local functions at the end of the parent method, even after return statements.
```

I prefer phrasing it this way as it is often alarming to people that these can be declared after `return` statements.